lib/streamlit/elements/widgets/chat.py

```diff
@@ -273,13 +271,13 @@ def chat_input(
             https://doc-chat-input.streamlit.app/
             height: 350px
 
-        The chat input can also be used inline instead of pinned to the
-        bottom by nesting it inside any other layout container
-        (container, columns, tabs, sidebar, etc).:
+        The chat input can also be used inline by nesting it inside any layout
+        container (container, columns, tabs, sidebar, etc). Create chat
+        interfaces embedded next to other content or have multiple chat bots!
 
         >>> import streamlit as st
         >>>
-        >>> with st.container():
+        >>> with st.sidebar:
```

Comment:
nit: maybe `container` is better here since it is the universal way how to move the chat input to inline, even within the main container. Since we haven't shipped the `position` parameter yet, this might otherwise not be very obvious. 

We might also want to turn this example to use a placeholder for writing back the prompt, since this will probably be the most common way on how it will be used. And this might also not be super obvious as you see in this issue: https://github.com/streamlit/streamlit/issues/8043

```
output = st.container()
prompt = st.chat_input("Say something")
if prompt:
    output.write(f"User has sent the following prompt: {prompt}")
```

Reply:
Or we leave the sidebar example as is and add one more example that's more chat-oriented:

```python
messages = st.container(height=300)
if prompt := st.chat_input("Say something"):
   messages.chat_message("user").write(prompt)
   messages.chat_message("assistant").write(f"Echo: {prompt}")

```

Reply:
Or we fully replicate the echo chat example with `session_state`

Reply:
I changed it from container to sidebar to make it more obvious. By just using container, it only floats to the top and isn't as obvious that it's different or inline. In the sidebar, it highlights that it can be moved around with the description above calling out the layout elements that can be used. I have another example I made which is a functionalized chat interface to make a complete, bordered chat interface with a bot that always replies with the same word.

```python
def chat_say_what(what):
    ...
```

I'm happy to change it back or add another example if that would be clearer. If the position argument is soon to follow, that would be a good third example.

Reply:
> By just using container, it only floats to the top and isn't as obvious that it's different or inline.

Ok, that makes sense. 

> I have another example I made which is a functionalized chat interface to make a complete

Will that be part of the `chat_input` page or the advanced chat page?

Reply:
In general, I think the most confusing part of the inline chat usage is the usage of the placeholder container to put in the messages. This isn't really an issue with the fullscreen mode since it is anyways always at the bottom, but for inline mode you usually want to place the input underneath the messages. But because of the execution logic, you only can add messages after having called `chat_input` -> and only way to do that is placeholder container.

Reply:
@LukasMasuch I switched it back to `st.container()`. When the position argument is added, we can refine further.

Reply:
The second comment was more related to making sure that the user understands how to correctly put elements above an inline `chat_input` widget. Keeping this in the sidebar for visual aspects makes sense to me, but maybe adding something like: 

```python
with st.sidebar:
    messages = st.container(height=300)
    if prompt := st.chat_input("Say something"):
       messages.chat_message("user").write(prompt)
       messages.chat_message("assistant").write(f"Echo: {prompt}")
 ```
 
 Or do you think the advanced examples will cover that well?